lms/djangoapps/shoppingcart/pdf.py

```diff
@@ -0,0 +1,466 @@
+        )
+        self.brand_logo_height = microsite.get_value(
+            "PDF_RECEIPT_LOGO_HEIGHT_MM", settings.PDF_RECEIPT_LOGO_HEIGHT_MM
+        ) * mm
+        self.cobrand_logo_height = microsite.get_value(
+            "PDF_RECEIPT_COBRAND_LOGO_HEIGHT_MM", settings.PDF_RECEIPT_COBRAND_LOGO_HEIGHT_MM
+        ) * mm
+
+        # From Context
+        self.items_data = context['items_data']
+        self.invoice_id = context['id']
+        self.date = ModuleI18nService().strftime(context['date'], 'SHORT_DATE')
+        self.is_invoice = context['is_invoice']
+        self.total_cost = '{currency}{amount:.2f}'.format(currency=self.currency, amount=context['total_cost'])
+        self.payment_received = '{currency}{amount:.2f}'.format(
+            currency=self.currency, amount=context['payment_received']
+        )
+        self.balance = '{currency}{amount:.2f}'.format(currency=self.currency, amount=context['balance'])
+
+        # initialize the pdf variables
+        self.margin = 15 * mm
+        self.page_width = letter[0]
+        self.page_height = letter[1]
+        self.min_clearance = 3 * mm
+        self.second_page_available_height = ''
+        self.second_page_start_y_pos = ''
+        self.first_page_available_height = ''
+        self.pdf = None
+
+    def is_on_first_page(self):
+        """
+        Returns True if it's the first page of the pdf, False otherwise.
+        """
+        return self.pdf.current_page_count() == 1
+
+    def generate_pdf(self, file_buffer):
+        """
+        Takes in a buffer and puts the generated pdf into that buffer.
+        """
+        self.pdf = NumberedCanvas(file_buffer, pagesize=letter)
+
+        self.draw_border()
+        y_pos = self.draw_logos()
+        self.second_page_available_height = y_pos - self.margin - self.min_clearance
+        self.second_page_start_y_pos = y_pos
+
+        y_pos = self.draw_title(y_pos)
+        self.first_page_available_height = y_pos - self.margin - self.min_clearance
+
+        y_pos = self.draw_course_info(y_pos)
+        y_pos = self.draw_totals(y_pos)
+        self.draw_footer(y_pos)
+
+        self.pdf.insert_page_break()
+        self.pdf.save()
+
+    def draw_border(self):
+        """
+        Draws a big border around the page leaving a margin of 15 mm on each side.
+        """
+        self.pdf.setStrokeColorRGB(0.5, 0.5, 0.5)
+        self.pdf.setLineWidth(0.353 * mm)
+
+        self.pdf.rect(self.margin, self.margin,
+                      self.page_width - (self.margin * 2), self.page_height - (self.margin * 2),
+                      stroke=True, fill=False)
+
+    @staticmethod
+    def load_image(img_path):
+        """
+        Loads an image given a path. An absolute path is assumed.
+        If the path points to an image file, it loads and returns the Image object, None otherwise.
+        """
+        try:
+            img = Image.open(img_path)
+        except IOError:
+            img = None
```

Comment:
Can we log out an error message to help any potential debugging sessions here? Thanks.